### PR TITLE
Add GitHub app

### PR DESCRIPTION
**Pull request by tom-mckay**
This PR:
- Adds a custom docker image built on the atlantis image.
- Updates the values files for staging and prod to support the two-stage deploy necessary for github app configuration

### CODE REVIEW

File: charts/atlantis/stages/production/values.yaml [hunk 1]
```diff
@@ -14,9 +14,48 @@ serviceAccount:
...
 environmentSecrets:
-  - name: ATLANTIS_WRITE_GIT_CREDS
+  - name: ATLANTIS_GH_WEBHOOK_SECRET
     secretKeyRef:
       name: atlantis-secret
-      key: atlantis_write_git_creds
+      key: github_token
+
+environment:
+  ATLANTIS_GH_APP_ID: 113650
```
Comment (rarkebauer) [inline charts/atlantis/stages/production/values.yaml]:
```suggestion
  ATLANTIS_GH_APP_ID: 113131
```

File: charts/atlantis/stages/production/values.yaml [hunk 2]
```diff
@@ -14,9 +14,48 @@ serviceAccount:
...
+  ATLANTIS_GH_ORG: procore
+
+github: null
+
+extraVolumes:
+  - name: gh-app-key-file
+    secret:
+      secretName: atlantis-secret
+      items:
+        - key: github_app_key_file
```
Comment (rarkebauer) [inline charts/atlantis/stages/production/values.yaml]:
```suggestion
        - key: github_key_file
```

File: charts/atlantis/stages/staging/values.yaml [hunk 3]
```diff
@@ -14,9 +14,49 @@ serviceAccount:
...
+  ATLANTIS_GH_ORG: procore
+
+github: null
+
+extraVolumes:
+  - name: gh-app-key-file
+    secret:
+      secretName: atlantis-secret
+      items:
+        - key: github_app_key_file
```
Comment (rarkebauer) [inline charts/atlantis/stages/staging/values.yaml]:
```suggestion
        - key: github_key_file
```

**Review by rarkebauer — CHANGES_REQUESTED** [review]

File: charts/atlantis/stages/production/values.yaml (hunk 2 above)
Comment (tom-mckay) [reply charts/atlantis/stages/production/values.yaml]: We were trying to keep the names of the keys in the secret somewhat aligned with the names of the environment variables they're used in. I'd actually prefer `atlantis_gh_app_key_file` for this one.

**Review by brandonpollard — APPROVED** [review]
lgtm

**Review by rarkebauer — APPROVED** [review]

**Pull request merged**